src/doc/reference.md

```diff
@@ -3163,7 +3163,7 @@ An example of a for loop over the contents of an array:
 
 let v: &[Foo] = &[a, b, c];
 
-for e in v.iter() {
+for e in v {
```

Comment:
Since `v` is already a reference, it appears that the for loop is actually using `v.iter()`. [This](http://is.gd/OaQtSw) code fails, but replacing it with `v.iter()` works.